common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -42,9 +42,23 @@ public static void cleanDirectBuffer(ByteBuffer buffer) {
     }
   }
 
+  /**
+   * Clone a bytebuffer.
+   * <p>
+   * The new bytebuffer will have the same content, but the cloning only applies to array-based
+   * bytebuffers
```

Comment:
what i meant is the cloned buffer may be a different type.. will change it to say that.